### ISSUE TITLE
Add some kind of script to automate following everyone

### ISSUE THREAD

**Issue by tylerl0706** [bnb/awesome-developer-streams#62]
Lots of great streamers! It'd be cool if there was a script to follow all of them at once :)

**Comment by bnb** [comment]
I'd not exactly be sure how to do this, especially since multiple devs stream on multiple platforms 😄

That said, if someone wanted to PR something I'd be TOTALLY cool with that 👍

**Comment by SCdF** [comment]
I was considering something like this, but I also use Twitch for other things and so it would be very noisy (ie I want to check these people out, but then it's hard to keep track of which ones I've seen and want to keep following, or aren't interested in and want to drop).

I think https://devstream.tv/directory is a pretty good initiative toward this kind of thing: one page where you can see everyone (on twitch at least) streaming.

I looked into Twitch communities to see if it was some kind of multi-reddit style thing where we could create a curated community which would give you one URL to check out, but it turns out that's not really what communities are, so never mind.

**Comment by grimmy** [comment]
Personally I think adding the data in a machine readable format that can be used to generate the `README.md` and then used for these other tools would be a good way to go.

**Comment by tylerl0706** [comment]
Expanding on the issue... The script could accept a `tags` parameter where you could specify "JavaScript" or "game dev" or whatever and it would filter this repo to just those folks and follow those.

I don't really use Twitch for anything else besides following developers so this sort of script was kinda what I was looking for :)

Totally agree on the machine readable approach generating human readable content. Allows for interesting scenarios that both machines and humans can consume :)

**Comment by bnb** [comment]
I'd be _totally_ open to PRs that add something like this in - would be especially awesome if there was some kind of CLI tool that existed to enter name/links/etc and it'd spit out the markdown and create the JSON.

That said, I don't (currently) have the time to build something like this myself 😞

**Comment by Tiim** [comment]
@bnb I started writing such a cli in node.js if you want to check it out on my fork in the [cli branch](https://github.com/Tiim/awesome-developer-streams/tree/cli) 😄 

I didn't write any documentation yet, but it supports adding streamers to a json file, generating the markdown README.md file from the json file and checking the json file schema for ci.

It's not yet ready for "production" but i would love to hear from you before I invest more time in this.

**Comment by tylerl0706** [comment]
Oh nice! So this looks like it's mainly for maintaining this repo, is that right? You'd run the cli to add a new streamer and then regenerate the markdown?

**Comment by Tiim** [comment]
Yeah exactly. You would add a streamer with the add command (not really user friendly but better for scripts and probably not even needed) or the interactive mode and then turn it to markdown with the render command.

If people want to use the data they can just download the `data.json` file from master and do whatever they want with it.

**Comment by tylerl0706** [comment]
That's sweet! Using json in code is so much better than parsing Markdown 😉

**Comment by RinaFave01** [comment]
Ok I think I'm caught up...
🤦‍♀️

**Comment by outfrost** [comment]
Now that people are using tags, you can bookmark a tag by bookmarking a link like this one:
https://www.twitch.tv/directory/all/tags/a59f1e4e-257b-4bd0-90c7-189c3efbf917
This one's for the `Programming` tag. You can't follow tags unfortunately, and to get these links you have to first go to a channel that's broadcasting with it and then click on that tag, but once you do it you have a button that can take you there with one click. I imagine maybe this repo or a related site could have buttons to jump straight to browsing by this and related tags on Twitch, such as `Game Development`, `Web Development`, etc.
